src/fusiontensor/base_interface.jl

```diff
@@ -0,0 +1,165 @@
+  new_data_matrix = data_matrix(left) - data_matrix(right)
+  return FusionTensor(
+    new_data_matrix, codomain_axes(left), domain_axes(left), trees_block_mapping(left)
+  )
+end
+
+function Base.:/(ft::FusionTensor, x::Number)
+  return FusionTensor(
+    data_matrix(ft) / x, codomain_axes(ft), domain_axes(ft), trees_block_mapping(ft)
+  )
+end
+
+Base.Array(ft::FusionTensor) = Array(cast_to_array(ft))
+
+# adjoint is costless: dual axes, swap codomain and domain, take data_matrix adjoint.
+# data_matrix coeff are not modified (beyond complex conjugation)
+transpose_mapping(d::Dict) = Dict([reverse(k) => transpose_mapping(v) for (k, v) in d])
+function transpose_mapping(b::BlockIndexRange{2})
+  new_block = Block(reverse(Tuple(Block(b))))
+  return new_block[reverse(b.indices)...]
+end
+function Base.adjoint(ft::FusionTensor)
+  return FusionTensor(
+    adjoint(data_matrix(ft)),
+    dual.(domain_axes(ft)),
+    dual.(codomain_axes(ft)),
+    transpose_mapping(trees_block_mapping(ft)),
+  )
+end
+
+Base.axes(ft::FusionTensor) = (codomain_axes(ft)..., domain_axes(ft)...)
+
+# conj is defined as coefficient wise complex conjugation, without axis dual
+Base.conj(ft::FusionTensor{<:Real}) = ft   # same object for real element type
+function Base.conj(ft::FusionTensor)
+  return FusionTensor(
+    conj(data_matrix(ft)), codomain_axes(ft), domain_axes(ft), trees_block_mapping(ft)
+  )
+end
+
+function Base.copy(ft::FusionTensor)
+  return FusionTensor(
+    copy(data_matrix(ft)),
+    copy.(codomain_axes(ft)),
+    copy.(domain_axes(ft)),
+    copy(trees_block_mapping(ft)),
+  )
+end
+
+function Base.deepcopy(ft::FusionTensor)
+  return FusionTensor(
+    deepcopy(data_matrix(ft)),
+    deepcopy.(codomain_axes(ft)),
+    deepcopy.(domain_axes(ft)),
+    deepcopy(trees_block_mapping(ft)),
+  )
+end
+
+# eachindex is automatically defined for AbstractArray. We do not want it.
+Base.eachindex(::FusionTensor) = error("eachindex not defined for FusionTensor")
+
+function Base.getindex(ft::FusionTensor, f1f2::Tuple{<:SectorFusionTree,<:SectorFusionTree})
+  return ft[f1f2...]
+end
+function Base.getindex(ft::FusionTensor, f1::SectorFusionTree, f2::SectorFusionTree)
+  charge_matrix = data_matrix(ft)[trees_block_mapping(ft)[f1, f2]]
+  return reshape(charge_matrix, charge_block_size(ft, f1, f2))
+end
+
+function Base.setindex!(
+  ft::FusionTensor, a::AbstractArray, f1f2::Tuple{<:SectorFusionTree,<:SectorFusionTree}
+)
+  return setindex!(ft, a, f1f2...)
+end
+function Base.setindex!(
+  ft::FusionTensor, a::AbstractArray, f1::SectorFusionTree, f2::SectorFusionTree
+)
+  return view(ft, f1, f2) .= a
+end
+
+Base.ndims(::FusionTensor{T,N}) where {T,N} = N
+
+Base.permutedims(ft::FusionTensor, args...) = fusiontensor_permutedims(ft, args...)
+
+function Base.similar(ft::FusionTensor)
+  mat = similar(data_matrix(ft))
+  initialize_allowed_sectors!(mat)
+  return FusionTensor(mat, codomain_axes(ft), domain_axes(ft), trees_block_mapping(ft))
+end
```

Comment:
I think you can just define this as:
```julia
Base.similar(ft::FusionTensor) = similar(ft, eltype(ft))
```

Reply:
If it's an AbstractArray, the default will always map it to `similar(ft, eltype(ft), axes(ft))`, so you actually would only have to define that.

Reply:
Good point, though maybe in this case we need both `similar(::FusionTensor, ::Type)` and `similar(::FusionTensor, ::Type, axes)`, since `axes(ft)` doesn't output bipartitioned axes (that also connects to my question about how we should define `axes(::FusionTensor)` in the first place...).

Reply:
So, related to the discussion elsewhere about the `axes` definition that was initiated by the `eachindex` definition, we should aim to output bipartitioned axes from `axes(::FusionTensor)`, and then only define `similar(::FusionTensor, ::Type, axes)` as suggested by @lkdvos, and then rely on the Base definitions of `similar(::FusionTensor)` and `similar(::FusionTensor, ::Type)` that handle the forwarding.